kaggle_environments/agent.py

```diff
@@ -134,14 +137,28 @@ def act(self, observation):
         ][:self.agent.__code__.co_argcount]
 
         # Start the timer.
-        start = perf_counter()
-        try:
-            action = self.agent(*args)
-        except Exception as e:
-            action = e
+
+        with StringIO() as out_buffer, StringIO() as err_buffer, redirect_stdout(out_buffer), redirect_stderr(err_buffer):
+            try:
+                start = perf_counter()
+                action = self.agent(*args)
+            except Exception as e:
+                traceback.print_exc(file=err_buffer)
+                action = e
+            # Allow up to 1k log characters per step which is ~1MB per 600 step episode
+            max_log_length = 1024
+            out = out_buffer.getvalue()[0:max_log_length]
+            err = err_buffer.getvalue()[0:max_log_length]
+
+        duration = perf_counter() - start
+        log = {
```

Comment:
Would it make sense to store `step` here?

Reply:
I don't think so because the index of the log in the array corresponds to step.